File: Java/Exercises/EventApp/app/src/main/java/com/anubys/example/eventapp/MainActivity.java
package com.anubys.example.eventapp;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ListAdapter;
import android.widget.ListView;

import com.anubys.example.eventapp.Modell.Event;
import com.google.firebase.database.ChildEventListener;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/** @Author Created by Anubys on the 07.04.2020 */


public class MainActivity extends AppCompatActivity {
    private static final String TAG = MainActivity.class.getSimpleName();

    @Nullable
    @BindView(R.id.lv_event_list)
    ListView listViewEvents;
    @Nullable
    @BindView(R.id.btn_add)
    Button buttonAdd;

    private Unbinder unbinder;
    private ArrayList<Event> arrayListEvent;
    private ListAdapter listAdapter;

    //* ************************************************ *
    //*               L I F E - C Y C L E                *
    //* ************************************************ *
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        Log.i(TAG, "TAG - MainActivity - onCreate()");
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        unbinder = ButterKnife.bind(this);

        getList();
        setListener();
    }

    @Override
    protected void onDestroy() {
        Log.i(TAG, "TAG - MainActivity - onDestroy()");
        super.onDestroy();

        if (unbinder != null) {
            unbinder.unbind();
        }
    }


    //* ************************************************ *
    //*         H E L P E R  -  M E T H O D S            *
    //* ************************************************ *
    private void setListener() {
        Log.i(TAG, "TAG - MainActivity - setListener()");
        if (buttonAdd != null) buttonAdd.setOnClickListener(v -> goToActivity());
    }

    private void goToActivity() {
        Log.i(TAG, "TAG - MainActivity - goToActivity()");
        Intent intent = new Intent(getApplicationContext(), AddEventActivity.class);
        startActivity(intent);
    }

    private void getList() {
        Log.i(TAG, "TAG - MainActivity - getList()");

        arrayListEvent = new ArrayList<>();
        listAdapter = new ArrayAdapter<>(getApplication(), android.R.layout.simple_list_item_1, arrayListEvent);

        // Datenbankverbindung aufbauen
        FirebaseDatabase firebaseDatabase = FirebaseDatabase.getInstance();
        // Den Pfad zu Events herstellen
        DatabaseReference databaseReference = firebaseDatabase.getReference("Events");
        // Die Änderungen des Wertes über Listener mitteilen und aktuallisieren
        databaseReference.addChildEventListener(new AddChildEventListener());
    }


    //* ************************************************ *
    //*                 C L A S S E S                    *
    //* ************************************************ *
    private class AddChildEventListener implements ChildEventListener {

        @Override
        public void onChildAdded(@NonNull DataSnapshot dataSnapshot, @Nullable String s) {
            Event event = dataSnapshot.getValue(Event.class);
            if (event != null) event.setKey(dataSnapshot.getKey());
            arrayListEvent.add(event);
            if (listViewEvents != null) {
                listViewEvents.setAdapter(listAdapter);
                listViewEvents.setOnItemClickListener((parent, view, position, id) -> {
                    Event eventClick = arrayListEvent.get(position);
                    Intent intent = new Intent(getApplicationContext(), DetailSiteActivity.class);
                    intent.putExtra("eventKey", eventClick.getKey());
                    startActivity(intent);
                });
            }
        }

        @Override
        public void onChildChanged(@NonNull DataSnapshot dataSnapshot, @Nullable String s) {

        }

        @Override
        public void onChildRemoved(@NonNull DataSnapshot dataSnapshot) {

        }

        @Override
        public void onChildMoved(@NonNull DataSnapshot dataSnapshot, @Nullable String s) {

        }

        @Override
        public void onCancelled(@NonNull DatabaseError databaseError) {

        }
    }
}

File: Kotlin/Examples/ViewBindingActivityDemo/app/src/main/java/com/anubys/example/viewbindingactivitydemo/MainActivity.kt
package com.anubys.example.viewbindingactivitydemo

/** @Author Created by Anubys on the 07.08.2020 */

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Toast

import androidx.appcompat.app.AppCompatActivity

import com.anubys.example.viewbindingactivitydemo.databinding.ActivityMainBinding


class MainActivity : AppCompatActivity() {
    private val tag = MainActivity::class.java.simpleName

    private var viewBinding: ActivityMainBinding? = null


    //* ************************************************ *
    //*               L I F E - C Y C L E                *
    //* ************************************************ *
    override fun onCreate(savedInstanceState: Bundle?) {
        Log.v(tag,"TAG - MainActivity - onCreate()")
        super.onCreate(savedInstanceState)
        viewBinding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(viewBinding?.root)

        setListener(viewBinding)
    }

    override fun onDestroy() {
        Log.v(tag,"TAG - MainActivity - onCreate()")
        super.onDestroy()
        viewBinding = null
    }


    //* ************************************************ *
    //*         H E L P E R  -  M E T H O D S            *
    //* ************************************************ *
    private fun setListener(viewBinding: ActivityMainBinding?) {
        Log.v(tag,"TAG - MainActivity - setListener()")
        viewBinding?.btnClickMe?.setOnClickListener{ clickMe(viewBinding) }
        viewBinding?.btnToToSubscribe?.setOnClickListener { gotToSubscribe() }
    }

    private fun clickMe(viewBinding: ActivityMainBinding?) {
        Log.v(tag,"TAG - MainActivity - clickMe()")
        Toast.makeText(applicationContext,"Clicked me", Toast.LENGTH_LONG).show()
        viewBinding?.tvShow?.text = viewBinding?.edtName?.text

    }

    private fun gotToSubscribe() {
        Log.v(tag,"TAG - MainActivity - gotToSubscribe()")
        Toast.makeText(applicationContext,"To to Subscribe", Toast.LENGTH_LONG).show()
        intent = Intent(applicationContext, SubscribeActivity::class.java)
        startActivity(intent)
    }
}
